docs/dg/dev/architecture/coding-conventions.md

```diff
@@ -0,0 +1,2165 @@
+
+There are four generic `Clients` as `SearchClient`, `SessionClient`, `StorageClient` and `ZedClient`.
+
+**Conventions**
+- The `Client` follows the [facade design pattern](#facade-design-pattern) and its conventions.
+
+**Example**
+```php
+namespace Spryker\Client\ConfigurableBundleCart;
+
+/**
+ * @method \Spryker\Client\ConfigurableBundleCart\ConfigurableBundleCartFactory getFactory()
+ */
+class ConfigurableBundleCartClient extends Spryker\Client\Kernel\AbstractClient implements ConfigurableBundleCartClientInterface
+{
+    /**
+     * {@inheritDoc}
+     *
+     * @api
+     *
+     * @param \Generated\Shared\Transfer\CreateConfiguredBundleRequestTransfer $createConfiguredBundleRequestTransfer
+     *
+     * @return \Generated\Shared\Transfer\QuoteResponseTransfer
+     */
+    public function addConfiguredBundle(CreateConfiguredBundleRequestTransfer $createConfiguredBundleRequestTransfer): QuoteResponseTransfer
+    {
+        return $this->getFactory()
+            ->createConfiguredBundleCartAdder()
+            ->addConfiguredBundleToCart($createConfiguredBundleRequestTransfer);
+    }
+}
+```
+
+```php
+namespace Spryker\Client\ConfigurableBundleCart;
+
+interface ConfigurableBundleCartClientInterface
+{
+    /**
+     * Specification:
+     * - Adds configured bundle to the cart.
+     * - Requires `configuredBundle.quantity` property to control amount of configured bundles put to cart.
+     * - Requires `configuredBundle.template.uuid` property to populate configurable bundle template related data.
+     * - Requires `items` property with `sku`, `quantity` and `configuredBundleItem.slot.uuid` properties to define how many
+     * items were added in total to a specific slot.
+     * - Returns QuoteResponseTransfer.
+     *
+     * @api
+     *
+     * @param \Generated\Shared\Transfer\CreateConfiguredBundleRequestTransfer $createConfiguredBundleRequestTransfer
+     *
+     * @return \Generated\Shared\Transfer\QuoteResponseTransfer
+     */
+    public function addConfiguredBundle(CreateConfiguredBundleRequestTransfer $createConfiguredBundleRequestTransfer): QuoteResponseTransfer;
+}
+```
+
+
+## Controller
+
+```
+[Organization]
+├── Zed
+│   └── [Module]
+│       └── Communication
+│           └── Controller
+│               ├── IndexController.php
+│               ├── GatewayController.php
+│               └── [Name]Controller.php
+│
+├── Yves
+│   └── [Module]
+│       └── Controller
+│           ├── IndexController.php
+│           └── [Name]Controller.php    
+│
+└── Glue
+    └── [Module]
+        └── Controller
+            ├── IndexController.php        
+            └── [Name]Controller.php                      
+```
+**Description**
+
+`Controllers` and `Actions` are application access points for any kind of HTTP communication with end-users or other applications.
+
+Responsibilities of a controller are
+- to adapt the received input data to the underlying layers (syntactical validation, delegation),
```

Comment:
Do we do any kind of validation in the controller?

```suggestion
- to pass the received input data to the underlying layers,
```

Reply:
We should :) 
https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller